RobotControl/Robot.cpp

```diff
@@ -149,11 +157,12 @@ void Robot::setStatus() {
   if(m_payload.getStatus() == STATUS_DISABLED) {
     return;
   }
+
   // Broken here
   if (m_firing) {
-    if(millis() - TEMP_FIRE_TIME_MILLIS < m_solenoidOpenMillis) {
+    if(millis() - m_fireTimeMillis < m_solenoidOpenMillis) {
```

Comment:
Personally, I'd set a time to close the valve when the button is pressed, and then check for:
 `millis() >= m_solenoidCloseMillis`